packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterInputMultipleSingleSelect.tsx

```diff
@@ -1,33 +1,16 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
-import Autocomplete, { AutocompleteProps, createFilterOptions } from '@mui/material/Autocomplete';
+import Autocomplete, { createFilterOptions } from '@mui/material/Autocomplete';
 import Chip from '@mui/material/Chip';
-
 import TextField from '@mui/material/TextField';
 import { unstable_useId as useId } from '@mui/material/utils';
-import { GridFilterItem } from '../../../models/gridFilterItem';
 import { getValueFromOption } from './filterPanelUtils';
+import { GridFilterInputMultipleSingleSelectProps } from './GridFilterInputMultipleSingleSelectProps';
 
-export type GridFilterInputMultipleSingleSelectProps = {
-  item: GridFilterItem;
-  applyValue: (value: GridFilterItem) => void;
-  // Is any because if typed as GridApiRef a dep cycle occurs. Same happens if ApiContext is used.
```

Comment:
I had to move `GridFilterInputMultipleSingleSelectProps` to a separate file to avoid circular dependency

Reply:
Even if you used `import type {}` you had a circular deps ? 
Normally `import type {}` never causes circular deps

Reply:
Ok, I cannot reproduce the issue anymore. Maybe it was eslint cache that was misleading